qpid-jms-examples/src/main/java/org/apache/qpid/jms/example/Server.java

```diff
@@ -0,0 +1,119 @@
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ *
+ */
+package org.apache.qpid.jms.example;
+
+import javax.jms.Connection;
+import javax.jms.ConnectionFactory;
+import javax.jms.DeliveryMode;
+import javax.jms.Destination;
+import javax.jms.ExceptionListener;
+import javax.jms.JMSException;
+import javax.jms.Message;
+import javax.jms.MessageConsumer;
+import javax.jms.MessageProducer;
+import javax.jms.Session;
+import javax.jms.TextMessage;
+import javax.jms.ObjectMessage;
+import javax.naming.Context;
+import javax.naming.InitialContext;
+import java.util.ArrayList;
+import java.util.Collections;
+
+public class Server {
+
+    public static void main(String[] args) throws Exception {
+	    	System.out.println("[SERVER] Awaiting Message...");
+	
+	        try {
+	            // The configuration for the Qpid InitialContextFactory has been supplied in
+	            // a jndi.properties file in the classpath, which results in it being picked
+	            // up automatically by the InitialContext constructor.
+	            Context context = new InitialContext();
+	
+	            ConnectionFactory factory = (ConnectionFactory) context.lookup("myFactoryLookup");
+	            Destination queue = (Destination) context.lookup("myQueueLookup");
+	
+	            Connection connection = factory.createConnection(System.getProperty("USER"), System.getProperty("PASSWORD"));
+	            connection.setExceptionListener(new MyExceptionListener());
+	            connection.start();
+	
+	            Session session = connection.createSession(false, Session.AUTO_ACKNOWLEDGE);
+	            
+	
+	            MessageConsumer messageConsumer = session.createConsumer(queue);
+	            while (true) {
+	            
+		            long start = System.currentTimeMillis();
+		            
+		            //Receive a message
+		            Message receivedMessage = messageConsumer.receive(0);
+		                
+		            long finish = System.currentTimeMillis();
+		            long taken = finish - start;
+	
+		            System.out.println("[SERVER] Received the message in " + taken + "ms");
+		            
+		            //Create new message to return to client
+		            Destination replyDestination = receivedMessage.getJMSReplyTo();
+		            TextMessage newMessage = session.createTextMessage(interpretMessage(receivedMessage));
+		            MessageProducer messageProducer = session.createProducer(null);
```

Comment:
MessageProducer instances need to be closed when you are finished using them, this would currently be orphaning some resources each time. You also typically create a long-lived anoynmous MessageProducers, so as to re-use them. If you move the creation outside the while loop, you can then reuse a single producer for sending all the subsequent responses.